classes/Site/Popups.php

```diff
@@ -110,7 +113,48 @@ public static function load_popups() {
 			// Clear the global $current.
 			pum()->current_popup = null;
 		}
+	}
+
+	/**
+	 * Checks post content to see if there are popups we need to automagically load
+	 *
+	 * @param string $content The content from the filter.
+	 * @return string The content.
+	 * @since 1.15
+	 */
+	public static function check_content_for_popups( $content ) {
+
+		// Only search for popups in the main query of a singular page.
+		if ( is_singular() && in_the_loop() && is_main_query() ) {
```

Comment:
@danieliser This is now only scanning during the main query of a singular post

Reply:
That is an improvement, but my original concern is that we are hooking filtered content passively, as opposed to scanning actively early in the head what `global $post->post_content` contains.

If this works for now, leave it, but I wasn't aware that apply_filters was called on the_content in the head before assets were enqueued.

Consider if they had no popups enqueued by default, and the_content isn't called until its rendered.

I'm not 100% that all assets get enqueued properly in that case, but I could be wrong entirely.

Again if its working as is, leave it for now.

Reply:
@danieliser None of your comments on this PR mention asset enqueueing so I must have missed where you mentioned that.

But, that is a good point to consider. Right now, all popups I have tested this with work as expected in my current solution. I'll test with a few different form plugins real quick before we move forward with this solution.

Reply:
@danieliser Tested with both Formidable Forms and Ninja Forms and both work correctly when inside a popup that was auto-enqueued.

Reply:
@fpcorso Well to be clear, during if no popups are enqueued, no assets get loaded. Was an early optimization.

The do_shortcode is processed right away when a popup is enqueued. So it happens before wp_enqueue_script in the head allowing any forms to fully enqueue on time.

But if its working for now, lets roll with it.